sklearn/mixture/tests/test_gmm.py

```diff
@@ -327,6 +327,26 @@ def test_1d_1component():
         assert_array_almost_equal(g.bic(X), g_full_bic)
 
 
+def test_fit_predict():
+    """
+    test that gmm.fit_predict works as expected
+    """
+    n_samples, n_dim, n_components = 100, 2, 2
+    mu = np.array([[8, 8]])
+    x0 = rng.randn(n_samples, n_dim)
+    x1 = rng.randn(n_samples, n_dim) + mu
+    x = np.vstack((x0, x1))
+
+    g_full = mixture.GMM(n_components=n_components, covariance_type='full',
+                         random_state=rng, min_covar=1e-7, n_iter=10)
+
+    predicts = g_full.fit_predict(x)
+
+    assert np.all(predicts[:n_samples] != predicts[n_samples:])
+    assert np.all(np.diff(predicts[:n_samples]) == 0)
+    assert np.all(np.diff(predicts[n_samples:]) == 0)
```

Comment:
I think it would be more explicit to use a cluster agreement score such as `adjusted_rand_score`:

```python
from sklearn.metrics.cluster import adjusted_rand_score

# ...
model = GMM(n_components=n_components, covariance_type='full',
                       random_state=rng, min_covar=1e-7, n_iter=10)
predictions_1 = model.fit(X).predict(X)
predictions_2 = model.fit_predict(X)

assert_equal(adjusted_rand_score(predictions_1, predictions_2), 1.0)
```

```